app/src/main/java/com/github/polybooks/database/FBInterestDatabase.kt

```diff
@@ -179,20 +212,116 @@ class FBInterestDatabase: InterestDatabase {
+                            }
+                            else -> {
+                                Semester(
+                                    section = interest["section"] as String,
+                                    semester = interest["semester"] as String
+                                )
+                            }
+                        }
+
+                    }
+                    future.complete(interestsList as List<Interest>)
+                } else {
+                    future.complete(emptyList())
+                }
+
+            }
+            .addOnFailureListener { future.completeExceptionally(DatabaseException("Could not retrieve the list of all interests of user ${user.uid} because of: $it")) }
+
+        return future
     }
 
     /**
-     * Sets the interests of the specified user.
+     * Sets the interests of the specified logged user.
      * TODO: Might need to add an authentication token to restrict authenticated users to only modify their interests.
      * @return A Future to receive confirmation of success/failure asynchronously
      * */
-    override fun setUserInterests(user: User, interests: List<Interest>): CompletableFuture<Unit> {
-        TODO("Not yet implemented")
+    fun setLoggedUserInterests(user: LoggedUser, interests: List<Interest>): CompletableFuture<List<Interest>> {
+        val future = CompletableFuture<List<Interest>>()
+
+        val docData = hashMapOf(
+            interestsTopicName to interests
+        )
+        FirebaseProvider.getFirestore()
+            .collection(userCollection)
+            .document(user.uid)
+            .set(docData, SetOptions.merge())
+            .addOnSuccessListener { future.complete(interests) }
+            .addOnFailureListener { future.completeExceptionally(DatabaseException("Failed to insert $docData of user ${user.uid} into Database because of: $it")) }
+
+        return future
+    }
+
+    // TODO list
+    // 2) use sharedPreferences for LoggedUser to serve as a cache.
+
+    /**
+     * Get the interests of the current unlogged local user
+     * As the user is not auth, it will use exclusively the local storage.
+     * */
+    private fun getLocalUserInterests(): CompletableFuture<List<Interest>> {
+        val future = CompletableFuture<List<Interest>>()
+        future.completeExceptionally(LocalUserException("user needs to be logged in to get their interests"))
+        return future
+    }
+
+    /**
+     * Sets the interests of the current unlogged local user.
+     * As the user is not auth, it will use exclusively the local storage.
+     * @return A Future to receive confirmation of success/failure asynchronously
+     * */
+    private fun setLocalUserInterests(interests: List<Interest>): CompletableFuture<List<Interest>> {
+        val future = CompletableFuture<List<Interest>>()
+        future.completeExceptionally(LocalUserException("user needs to be logged in to set their interests to $interests"))
```

Comment:
Will you implement this in a future PR?

Reply:
I intended to initially,
but then, I discussed that with Joshua, and his UI currently do not allow any unlogged (= local) user to set their interests, and he doesn't intend on changing that before the deadline, so there's no point in pre-coding the back-end for it, given we have two days left.

Reply:
I can only agree with that